BlueShift-iOS-SDK/BlueShift.m

```diff
@@ -42,113 +58,122 @@ - (void)setAppDelegate {
-    }
-    
-    // Initialize deeplinks ...
-    [self initDeepLinks];
-    
-    // Getting the original Delegate ...
-    NSObject<UIApplicationDelegate> *oldDelegate = [UIApplication sharedApplication].delegate;
-    
-    // initiating the newDelegate ...
-    _newDelegate = [[BlueShiftAppDelegate alloc] init];
-    
-    // Initialise core data
-    [_newDelegate initializeCoreData];
-    BlueShiftUserNotificationCenterDelegate *blueShiftUserNotificationCenterDelegate = [[BlueShiftUserNotificationCenterDelegate alloc] init];
-    
-    // assigning the current application delegate with the app delegate we are going to use in the SDK ...
-    _sharedBlueShiftInstance.appDelegate = _newDelegate;
-    _sharedBlueShiftInstance.userNotificationDelegate = blueShiftUserNotificationCenterDelegate;
-    
-    // setting the new delegate's old delegate with the original delegate we saved...
-    BlueShiftAppDelegate *blueShiftAppDelegate = (BlueShiftAppDelegate *)_newDelegate;
-    blueShiftAppDelegate.oldDelegate = oldDelegate;
-    
-    if (@available(iOS 10.0, *)) {
-        if(config.userNotificationDelegate) {
-            blueShiftAppDelegate.userNotificationDelegate = config.userNotificationDelegate;
-        } else {
-            blueShiftAppDelegate.userNotificationDelegate = blueShiftUserNotificationCenterDelegate;
+    @try {
+        // Validate the API key
+        if ([config validateConfigDetails] == NO) {
+            return ;
         }
-    }
-    
-    if(config.blueShiftPushDelegate) {
-        blueShiftAppDelegate.blueShiftPushDelegate = config.blueShiftPushDelegate;
-    }
-    if (config.blueshiftUniversalLinksDelegate) {
-        blueShiftAppDelegate.blueshiftUniversalLinksDelegate = config.blueshiftUniversalLinksDelegate;
-    }
-    if (config.blueshiftDeviceIdSource) {
-        [[BlueShiftDeviceData currentDeviceData] setBlueshiftDeviceIdSource:config.blueshiftDeviceIdSource];
         
-        //Custom device id provision for DeviceIDSourceCUSTOM
-        if (config.blueshiftDeviceIdSource == BlueshiftDeviceIdSourceCustom) {
-            if (config.customDeviceId && ![config.customDeviceId isEqualToString:@""]) {
-                [[BlueShiftDeviceData currentDeviceData] setCustomDeviceID:config.customDeviceId];
-                [BlueshiftLog logInfo: [NSString stringWithFormat:@"CUSTOM device id is set as - %@",config.customDeviceId] withDetails:nil methodName:nil];
+        // Set config
+        _sharedBlueShiftInstance.config = config;
+        
+        // Set up device id
+        if (config.blueshiftDeviceIdSource) {
+            [[BlueShiftDeviceData currentDeviceData] setBlueshiftDeviceIdSource:config.blueshiftDeviceIdSource];
+            
+            //Custom device id provision for DeviceIDSourceCUSTOM
+            if (config.blueshiftDeviceIdSource == BlueshiftDeviceIdSourceCustom) {
+                if (config.customDeviceId && ![config.customDeviceId isEqualToString:@""]) {
```

Comment:
we had some utility method that does this nil and empty check, right? can we use that here?

Reply:
Yes, I will make use of it.